lib/Conversion/StandardToHandshake/StandardToHandshake.cpp

```diff
@@ -1667,6 +1667,60 @@ struct HandshakeAnalysisPass
 };
 } // namespace
 
+// Temporary: remove unused function args
+// The array arguements are impelemented as local memory ops in handshake
+// dialect, and the original arguements have no use. This pass temporary removes
+// these used args to let the simulator work properly by reading external data
+// files.
+namespace {
+
+struct HandshakeCleanFuncArgPass
+    : public HandshakeCleanFuncArgBase<HandshakeCleanFuncArgPass> {
+  void runOnOperation() override {
+    auto m = getOperation();
+    for (auto funcOp : m.getOps<handshake::FuncOp>()) {
```

Comment:
should this pass run on handshake::FuncOp's instead of ModuleOp?

Reply:
+1 I agree.